src/pool.c

```diff
@@ -77,18 +95,26 @@ int pool_init(int size)
 {
    int err;
 
+   spin_lock(&pool_lock);
```

Comment:
We don't need this `spin_lock()`

Please rewrite the pool API to reflect that each pool should be associated with a listener, I completely missed this part here.

Ideally we should move the `pool_head`, `pool_lock` and `pool_size` to `struct context`